frontend2/src/components/EpisodeLayout.tsx

```diff
@@ -9,9 +9,11 @@ import { EpisodeContext } from "../contexts/EpisodeContext";
 const EpisodeLayout: React.FC = () => {
   const episodeContext = useContext(EpisodeContext);
   const { episodeId } = useParams();
-  if (episodeId !== undefined && episodeId !== episodeContext.episodeId) {
-    episodeContext.setEpisodeId(episodeId);
-  }
+  useEffect(() => {
+    if (episodeId !== undefined && episodeId !== episodeContext.episodeId) {
+      episodeContext.setEpisodeId(episodeId);
+    }
+  }, [episodeId]);
```

Comment:
Note: I put this in a useeffect so react doesn't yell at us

Reply:
Wheee luv react 😃